mlflow/types/type_hints.py

```diff
@@ -342,7 +367,9 @@ def _infer_schema_from_list_type_hint(type_hint: type[list[Any]]) -> Schema:
     """
     if not _is_list_type_hint(type_hint):
         raise InvalidTypeHintException(
-            message=f"Type hint for model input must be `list[...]`, got {type_hint}",
+            message="Type hints must be wrapped in list[...] to support "
+            "multiple-record batch inference. Specify your type hint as "
```

Comment:
```suggestion
            message="Type hints must be wrapped in list[...] because MLflow assumes the "
            "predict method to take multiple input instances. Specify your type hint as "
```

nit: `to support multiple-record batch inference` might be misleading, like "oh I don't need batch inference support".